pkg/autodiscovery/autoconfig.go

```diff
@@ -495,6 +454,55 @@ func (ac *AutoConfig) pollConfigs() {
 	}()
 }
 
+// TODO: move to check scheduler
+func (ac *AutoConfig) processRemovedConfigs(removedConfigs []integration.Config) {
+	// Process removed configs first to handle the case where a
+	// container churn would result in the same configuration hash.
+	for _, config := range removedConfigs {
+		if !isCheckConfig(config) {
+			// skip non check configs.
+			continue
+		}
+		// unschedule all the checks corresponding to this config
+		digest := config.Digest()
+		ids := ac.configToChecks[digest]
+		stopped := map[check.ID]struct{}{}
+		for _, id := range ids {
+			// `StopCheck` might time out so we don't risk to block
+			// the polling loop forever
+			err := ac.collector.StopCheck(id)
+			if err != nil {
+				log.Errorf("Error stopping check %s: %s", id, err)
+				errorStats.setRunError(id, err.Error())
+			} else {
+				stopped[id] = struct{}{}
+			}
+		}
+
+		// remove the entry from `config2checks`
```

Comment:
let's update the comments too: `configToChecks`

Reply:
I also renamed `name2jmxmetrics` ➡️ `nameToJMXMetrics` while at it